pkg/tests/apis/iam/iam_test.go

```diff
@@ -173,3 +176,142 @@ func TestIntegrationIdentity(t *testing.T) {
 		]`, found)
 	})
 }
+
+func TestIntegrationUsers(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	t.Run("User CRUD operations with dual writer mode 0", func(t *testing.T) {
+		helper := apis.NewK8sTestHelper(t, testinfra.GrafanaOpts{
+			AppModeProduction:    false,
+			DisableAnonymous:     true,
+			APIServerStorageType: "unified",
+			UnifiedStorageConfig: map[string]setting.UnifiedStorageConfig{
+				"users.iam.grafana.app": {
+					DualWriterMode: grafanarest.Mode0,
+				},
+			},
+			EnableFeatureToggles: []string{
+				featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs,
+			},
+		})
+		doUserCRUDTests(t, helper)
+	})
+
+	t.Run("User CRUD operations with dual writer mode 1", func(t *testing.T) {
```

Comment:
Could you expand these tests to add modes 2-4?

Reply:
Yes, I added the test for modes between 0-3, but skipped mode4 for now and we'll fix that in a follow up PR.